localization/ekf_localizer/CMakeLists.txt

```diff
@@ -46,7 +47,8 @@ if(BUILD_TESTING)
     test/test_mahalanobis.cpp
     test/test_measurement.cpp
     test/test_numeric.cpp
-    test/test_state_transition.cpp)
+    test/test_state_transition.cpp
+    test/test_warning_message.cpp)
```

Comment:
IMHO this test is better not to be added, either.

This is too dependent on the implementation, and thus would increase a cost of modifying them (e.g. changing the error messages). Also, I see little advantage of adding this test, since it is essentially a test for `fmt::format`.

What do you think?

Reply:
(Same argument applies to https://github.com/autowarefoundation/autoware.universe/pull/2694)

Reply:
Let's say we modify the warning message in the future, and with some mistake we forget to remove `{}` in the format string.

```cpp
const std::string s = "The delay step {} should be less than the maximum state step {}. Some forgotten comment {}";
```

And then we run this code, C++ throws the error at the **runtime**, not the compile time.

```
C++ exception with description "argument not found" thrown in the test body.
```

To avoid this runtime error, we need this test.

Reply:
I see, that make sense. Thank you :+1: